src/panorama/VideoPanorama.js

```diff
@@ -31,6 +34,17 @@
 
 		this.videoFramerate = 30;
 
+		function isSafari10 () {
+			var ua=navigator.userAgent,tem,M=ua.match(/(opera|chrome|safari|firefox|msie|trident(?=\/))\/?\s*(\d+)/i) || [];
+
+			M=M[2]? [M[1], M[2]]: [navigator.appName, navigator.appVersion, '-?'];
+			if((tem=ua.match(/version\/(\d+)/i))!=null) {M.splice(1,1,tem[1]);}
+			if (M[0] === "Safari") {
+				return parseInt(M[1]) >= 10;
+			} else return false;
+		}
```

Comment:
Pretty concise regex. I'd hope it can be reformatted to align with style, please bear with me. :) I'll setup another refactor to merge this with `this.isIOS` to make them more synchronized.
```
function isIOS10 () {
    var ua = navigator.userAgent,
          M = ua.match(/(opera|chrome|safari|firefox|msie|trident(?=\/))\/?\s*(\d+)/i) || [],
          tem;
    M = M[ 2 ] ? [ M[ 1 ], M[ 2 ] ] : [ navigator.appName, navigator.appVersion, '-?' ];
    if ( ( tem = ua.match(/version\/(\d+)/i)) !== null ) {
        M.splice( 1, 1, tem[ 1 ] );
    }
    return ( M[ 0 ] === "Safari" ? parseInt( M[ 1 ] ) >= 10 : false )
}
```